src/components/quiz-question/QuizScreen.tsx

```diff
@@ -0,0 +1,160 @@
+import React, {useState, useEffect} from 'react'
+import { Navigate, useParams, useNavigate } from 'react-router-dom';
+import { QuizQuestions } from './QuizQuestions';
+import { supabase } from '../../../supabaseClient';
+import { useToast, useQuizPlay, useAuth} from '../../contexts';
+import {quizQuestionType, quizDataType, quizCategoryDetailsType} from './quizQuestionTypes';
+
+import { useAchievementDetail } from '../../hook';
+
+const QuizScreen = () => {
+
+    const [quizQuestions, setQuizQuestions] = useState<quizQuestionType[]>([]);
+    const [quizDetails, setQuizDetails] = useState<quizCategoryDetailsType>();
+    const [questionCurrentIndex, setQuestionCurrentIndex] = useState(0);
+    const [markedAnswers, setMarkedAnswers] = useState<Array<number|null|undefined>>(new Array(quizQuestions.length));
+    
+    let quizScoreResult =0;
+    let correctAnswers = 0;
+    let percentage =0;
+    let isQuizPassed : boolean;
+
+   const isQuenstionsEnd = (questionCurrentIndex === quizQuestions.length);
+  
+    const {playquizid} = useParams();
+    const {notifyError} = useToast();
+    const navigate =  useNavigate();
+    const { userDetails, getUserDetails, setUserDetails} = useAchievementDetail();
+    //const {isQuizStarted, setIsQuizStarted} = useQuizPlay();
+    const {currentUser} =  useAuth();
+
+
+  const getQuizQuestions = async ()=>{
+    
+    try{
+    const { data, error } : quizDataType = await supabase.from('category_details')
+     .select('*, category!inner(*)')
+     .eq('category.quizid', playquizid)
+           
+      if(data !== null){
+        setQuizQuestions(data[0].quiz_option_answer)
+        setQuizDetails(data[0].category)
+      }
+      if (error) throw error
+    }catch(error){
+      notifyError(error.message);
+      console.error(error)
+    }
+
+}
+
+    useEffect(()=>{
+     
+      getQuizQuestions()
+    },[])
+
+    useEffect(()=>{
+      if(quizQuestions.length > 0){
+
+        if(isQuenstionsEnd){
+        
+         quizScoreResult = quizQuestions.reduce((acc, curr,index) => {
+        if(curr.answerIndex === markedAnswers[index]){
+          correctAnswers ++;
+            acc = acc+ curr.points;
+        }
+        return acc;
+       }, 0)
+
+       
+        if(quizScoreResult * (100/ quizDetails?.quiz_points!) > 50){
```

Comment:
Appreciation: you have used "bang operator" for some code reduction. Great work.